x-pack/plugins/ingest_manager/server/plugin.ts

```diff
@@ -92,6 +117,25 @@ export class IngestManagerPlugin implements Plugin {
     // TODO: Use this.config$ + if security is enabled to register conditional routing
     registerEPMRoutes(router);
     registerFleetSetupRoutes(router);
+    registerAgentRoutes(router);
+    registerEnrollmentApiKeyRoutes(router);
+
+    return {
+      __legacyCompat: {
+        registerLegacyAPI: (legacyAPI: LegacyAPI) => {
+          this.legacyAPI = legacyAPI;
+          const {
+            SavedObjectsClient,
+            getSavedObjectsRepository,
+          } = this.getLegacyAPI().savedObjects;
+          const { callAsInternalUser } = this.clusterClient;
+          const internalRepository = getSavedObjectsRepository(callAsInternalUser);
+
+          const internalSavedObjectsClient = new SavedObjectsClient(internalRepository);
+          appContextService.setInternalSavedObjectsClient(internalSavedObjectsClient);
+        },
+      },
+    };
   }
 
   public async start(
```

Comment:
an internally-scoped SO client can be created using `core` provided to this plugin `start()`:

```
import { SavedObjectsClient } from 'kibana/server';

...

const internalSavedObjectsClient = new SavedObjectsClient(core.savedObjects.createInternalRepository());

...

appContextService.start({
  internalSavedObjectsClient,
  ...
});
```